src/SFML/Audio/SoundRecorder.cpp

```diff
@@ -202,6 +213,32 @@ const std::string& SoundRecorder::getDevice() const
 
 
 ////////////////////////////////////////////////////////////
+void SoundRecorder::setChannelCount(unsigned int channelCount)
+{
+    if (m_isCapturing)
+    {
+        sf::err() << "It's not possible to change the channels while recording. Please stop the recording and call setChannelCount() again." << std::endl;
+        return;
+    }
+
+    if (channelCount > 2)
```

Comment:
I was talking about this implementation here, which currently allows you to use a channel count of 0, which is undefined and will lead to who knows what. Thus the "range" I talked about would be `if(channelCount < 1 && channelCount > 2)` or similar.

If you want to suggest a different design, you might want to put that more clearly somewhere (else). :wink: